File: python/fundamentals/day-02/Student_Result_System.py
def get_grade_and_result(marks):
    if marks >= 90:
        return "A", "PASS"
    elif marks >= 80:
        return "B", "PASS"
    elif marks >= 70:
        return "C", "PASS"
    elif marks >= 60:
        return "D", "PASS"
    elif marks >= 50:
        return "E", "PASS"
    else:
        return "F", "FAIL"

def main():
    # Input
    name = input("Enter Name: ")
    marks = float(input("Enter Marks: "))
    
    # Process
    grade, result = get_grade_and_result(marks)
    
    # Display Output
    print("\n-------------")
    print("Student Report\n")
    print(f"Name   : {name}")
    print(f"Marks  : {int(marks) if marks.is_integer() else marks}")
    print(f"Grade  : {grade}")
    print(f"Result : {result}")

if __name__ == "__main__":
    main()

File: FullStack_Java_Roadmap/src/fundamentals/day02/StudentResultSystem.java
package fundamentals.day02;
import java.util.Scanner;
public class StudentResultSystem {
	public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input
        System.out.print("Enter Name: ");
        String name = scanner.nextLine();

        System.out.print("Enter Marks: ");
        double marks = scanner.nextDouble();

        // Process Grade and Result
        String grade;
        String result;

        if (marks >= 90) {
            grade = "A";
            result = "PASS";
        } else if (marks >= 80) {
            grade = "B";
            result = "PASS";
        } else if (marks >= 70) {
            grade = "C";
            result = "PASS";
        } else if (marks >= 60) {
            grade = "D";
            result = "PASS";
        } else if (marks >= 50) {
            grade = "E";
            result = "PASS";
        } else {
            grade = "F";
            result = "FAIL";
        }

        // Display Output
        System.out.println("\n-------------");
        System.out.println("Student Report\n");
        System.out.println("Name   : " + name);
        if (marks % 1 == 0) {
            System.out.println("Marks  : " + (int) marks);
        } else {
            System.out.println("Marks  : " + marks);
        }
        System.out.println("Grade  : " + grade);
        System.out.println("Result : " + result);

        scanner.close();
    }
}
